superduperdb/backends/ibis/cdc/listener.py

```diff
@@ -0,0 +1,203 @@
+import threading
+import typing as t
+import time
+
+
+from superduperdb import logging
+from superduperdb.backends.ibis import query
+from superduperdb.cdc import cdc
+from superduperdb.misc.runnable.runnable import Event
+
+if t.TYPE_CHECKING:
+    from superduperdb.base.datalayer import Datalayer
+
+
+class PollingStrategyIbis:
+    def __init__(self, db: 'Datalayer', table: 'Table', options: t.Dict):
+        self.db = db
+        self.table = table
+        self.options = options
+
+        self.id = options.get('primary_id', 'id')
+        self.increment_field = options.get('auto_increment_field', None)
+        self.frequency = options.get('frequency', 3600)
+        self._last_processed_id = -1
+
+    def fetch_ids(self):
+        return ()
+
+    def post_handling(self):
+        time.sleep(self.frequency)
+
+    def get_strategy(self):
+        if self.increment_field:
+            return PollingStrategyIbisByIncrement(self.db, self.table, self.options)
+        else:
+            return PollingStrategyIbisByID(self.db, self.table, self.options)
+
+class PollingStrategyIbisByIncrement(PollingStrategyIbis):
+    def fetch_ids(self, ):
+        assert self.increment_field
+        increment_field = getattr(self.table, self.increment_field)
+        query = self.table.select(self.id).filter( increment_field > self._last_processed_id)
+        ids = list(self.db.execute(query))
+        ids =  [id[self.id] for id in ids]
+        self._last_processed_id += len(ids)
+        return ids
+
+class PollingStrategyIbisByID(PollingStrategyIbis):
+    ...
+
+
+
+class IbisDatabaseListener(cdc.BaseDatabaseListener):
+    """
+    It is a class which helps capture data from ibis database and handle it
+    accordingly.
+
+    This class accepts options and db instance from user and starts a scheduler
+    which could schedule a listening service to listen change stream.
+
+    This class builds a workflow graph on each change observed.
+
+    """
+
+    DEFAULT_ID: str = 'id'
+    EXCLUSION_KEYS: t.Sequence[str] = [DEFAULT_ID]
+    IDENTITY_SEP: str = '/'
+    _scheduler: t.Optional[threading.Thread]
+
+    def __init__(
+        self,
+        db: 'Datalayer',
+        on: query.Table,
+        stop_event: Event,
+        identifier: 'str' = '',
+        timeout: t.Optional[float] = None,
+        strategy: t.Dict = {'strategy': 'polling', 'options': {'frequency': 3600, 'auto_increment_field': None}}
```

Comment:
How does this get injected into the class? From `CFG`?

Reply:
uhm .. good question 
no its not injected in CFG